### ISSUE TITLE
6.1.71: InMemoryAppender errors when running docker-compose up

### ISSUE THREAD

**Issue by glc650** [jacobalberty/unifi-docker#423]
### Host operating system

Ubuntu 20.04.2 LTS

### What tag are you using

jacobalberty/unifi:${TAG:-latest}

### What complete docker command or docker-compose.yml do you use to launch the container (omitting sensitive values)?

version: '3.8'
services:
  controller:
    image: "jacobalberty/unifi:${TAG:-latest}"
    container_name: unifi
    init: true
    restart: unless-stopped
    volumes:
      - /home/server/docker/unifi/config:/unifi
    environment:
      - RUNAS_UID0=false
      - UNIFI_UID=1000
      - UNIFI_GID=1000
      - TZ=America/Los_Angeles
      - JVM_MAX_THREAD_STACK_SIZE=1280k
    # ports:
      # - "10.0.0.4:3478:3478/udp" # STUN
      # - "10.0.0.4:6789:6789/tcp" # Speed test
      # - "10.0.0.4:8080:8080/tcp" # Device/ controller comm.
      # - "10.0.0.4:8443:8443/tcp" # Controller GUI/API as seen in a web browser
      # - "10.0.0.4:10001:10001/udp" # AP discovery
    network_mode: "host"

### What do you expect to happen?

No errors when running docker-compose up

### What actually happens?

These errors after running docker-compose up (which was run after running docker stop unifi, docker rm unifi, docker system prune --all --volumes):

unifi         | 2021-03-27 10:23:40,501 main ERROR Error processing element InMemoryAppender ([Appenders: null]): CLASS_NOT_FOUND
unifi         | 2021-03-27 10:23:40,731 main ERROR Unable to locate appender "InMemoryAppender" for logger config "root"

The container DOES seem to function properly, however.

**Comment by josmellop** [comment]
Same problem here since last update, and the container seems to function properly too.
Someone could help us?

**Comment by foorschtbar** [comment]
> 
> 
> Same problem here since last update, and the container seems to function properly too.
> In my case, I'm running it on a Raspberry Pi 4.
> 

Same here:

```
unifi-controller | 2021-03-28 16:25:48,912 main ERROR Error processing element InMemoryAppender ([Appenders: null]): CLASS_NOT_FOUND
unifi-controller | 2021-03-28 16:25:50,088 main ERROR Unable to locate appender "InMemoryAppender" for logger config "root"

```

**Comment by JCBird1012** [comment]
From everything I can see while investigating (I also experience the same issue) - this appears to be an issue upstream with the controller binary itself. I don't think there's necessarily any indication that it is related to running it in a Docker container or anything with jacobalberty's image. 

https://gist.github.com/davecoutts/5ccb403c3d90fcf9c8c4b1ea7616948d
In the comments of the gist above, you can see people experiencing the same errors, and they're running it directly on the host (and not in containers)

@jacobalberty - this can likely be closed as it's not actionable for this repository at this time.

**Comment by jacobalberty** [comment]
> 
> Downgrade to image/tag: `jacobalberty/unifi:6.0` also not working, it seams that jacobalberty pushed to the wrong branch? Docker Hub shows that the 6.0-Tag had also the URL to the 6.1.x-deb file (https://hub.docker.com/layers/jacobalberty/unifi/6.0/images/sha256-d812ba4c27bd544fe11fab36d8c66783c28dd1ac6d87b9f106bdf6ba9919a94f?context=explore).

I forgot to pull the tag for 6.0 off of travis, i've updated the tags now i'll push 6.0.45 back out to 6.0 and new builds will go into 6.1

**Comment by zomborro** [comment]
Running this container on my Synology. However using the Lastest image, I know receive the same log errors as mentioned by @foorschtbar. The container will start, however I'm not able again to open the Unifi controller page as I'll reveive a 'HTTP Status 404 – Not Found'.


2021-04-17 12:00:38 | stdout | Caused by: java.lang.RuntimeException: Web context failed to initialize
-- | -- | --
2021-04-17 12:00:38 | stdout | at com.ubnt.ace.Launcher.main(Unknown Source)
2021-04-17 12:00:38 | stdout | at com.ubnt.service.ooOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO.Óo0000(Unknown Source)
2021-04-17 12:00:38 | stdout | at com.ubnt.net.OoOO.if(Unknown Source)
2021-04-17 12:00:38 | stdout | Exception in thread "launcher" java.lang.IllegalStateException: Tomcat failed to start up
2021-04-17 12:00:14 | stdout | 2021-04-17 12:00:14,685 main ERROR Unable to locate appender "InMemoryAppender" for logger config "root"
2021-04-17 12:00:13 | stdout | 2021-04-17 12:00:13,311 main ERROR Error processing element InMemoryAppender ([Appenders: null]): CLASS_NOT_FOUND
2021-04-17 12:00:10 | stdout | [2021-04-17 12:00:10,663] <docker-entrypoint> Starting unifi controller service.


How to fix this? Please be aware I'm a bit of a Docker noob here...

In addition I also downloaded the 6.0.45 image. Is there a way to connect the 'Latest' container to the 6.0.45 image?

**Comment by foorschtbar** [comment]
Is there any plan to fix 6.1er container? 6.0.45 works, but when i switch to tag `latest `or `stable-6` i still got the `Error processing element InMemoryAppender ([Appenders: null]): CLASS_NOT_FOUND` error...

**Comment by jacobalberty** [comment]
@foorschtbar the only fix would be to use 6.2.x, there's a version in rc right now. 6.1 runs fine for me in both of my  installs but gives that error message as well so I believe that error message is a red herring and many people seeing it just need to wait a bit longer for the controller to start. However others it does appear to not be coming up at all, regardless I believe that error message is unrelated to why it isn't starting.

It does not appear ubiquiti will be publishing a fix for 6.1.x as they have already moved on to 6.2.x

**Comment by AndyHerb** [comment]
Having followed the steps in [this article](https://help.ui.com/hc/en-us/articles/205144550-UniFi-Run-the-Controller-as-a-Windows-Service) to install the controller as a service on Windows 10 (20H2 19042.928) I'm getting the same error as others in this thread.  Although it's created the service I got the below errors when attempting to start the service via the command line:

`c:\Users\<my user>\Ubiquiti UniFi>java -jar lib\ace.jar startsvc
2021-05-04 13:40:06,549 main ERROR Error processing element InMemoryAppender ([Appenders: null]): CLASS_NOT_FOUND
2021-05-04 13:40:06,803 main ERROR Unable to locate appender "InMemoryAppender" for logger config "root"`

When attempting to start the service via the command line or Service Manager I get the following entries in the System event log:

- Event ID: 7045
- Level: Error
- Source: Service Control Manager
- Description:
> The UniFi Controller service terminated with the following service-specific error: 
> Incorrect function.

I expect it's entirely the same problem as has been logged already, but hopefully some of the above will help someone else find this quicker next time.

**Comment by mattbsea** [comment]
I had this issue after upgrading to 6.1.71.  The root cause was a wifi schedule conflict that was crashing the controller.  A description and fix are documented here:

https://community.ui.com/questions/UDM-Edit-WLAN-schedule-via-SSH/f68c0dac-7f4d-406b-8b7c-fe2598b6fb58#answer/f75ae132-b7ba-44ed-a74c-0eb2414736fe

**Comment by foorschtbar** [comment]
> 
> 
> @foorschtbar the only fix would be to use 6.2.x, there's a version in rc right now. 6.1 runs fine for me in both of my installs but gives that error message as well so I believe that error message is a red herring and many people seeing it just need to wait a bit longer for the controller to start. However others it does appear to not be coming up at all, regardless I believe that error message is unrelated to why it isn't starting.
> 
> It does not appear ubiquiti will be publishing a fix for 6.1.x as they have already moved on to 6.2.x

Thank you @jacobalberty for the quick reply. But, I still don't understand what options I have now. Is there anything I can do NOW to have a working 6.2.x container (use a special tag) or should i wait?
